src/tools/components/InputParameters.tsx

```diff
@@ -8,43 +7,73 @@ import { UseDataAPIState } from '../../shared/hooks/useDataApi';
 
 import { PublicServerParameters } from '../types/toolsServerParameters';
 import { JobTypes } from '../types/toolsJobTypes';
+import { FinishedJob } from '../types/toolsJob';
 
 type InputParametersProps = {
   id: string;
-  inputParamsData: Partial<UseDataAPIState<PublicServerParameters[JobTypes]>>;
+  // No public endpoint to expose this for peptide search, so for now replace
+  // with a "possible" job object in the case it's the same user that created it
+  inputParamsData:
+    | Partial<UseDataAPIState<PublicServerParameters[JobTypes]>>
+    | FinishedJob<JobTypes.PEPTIDE_SEARCH>
+    | null;
   jobType: JobTypes;
 };
 
-const InputParameters: FC<InputParametersProps> = ({
+const fieldsToHide = new Set(['redirectURL']);
+
+const InputParameters = ({
   id,
   inputParamsData,
   jobType,
-}) => {
-  const { loading, data, error, status } = inputParamsData;
+}: InputParametersProps) => {
+  if (inputParamsData && !('type' in inputParamsData)) {
+    if (inputParamsData.error || !inputParamsData.data) {
+      return <ErrorHandler status={inputParamsData.status} />;
+    }
+  }
 
-  if (error || !data) {
-    return <ErrorHandler status={status} />;
+  if (!inputParamsData) {
+    if (jobType === JobTypes.PEPTIDE_SEARCH) {
+      return (
+        <section>
+          <p>
+            Unable to retrieve the input values for the job with ID{' '}
+            <code>{id}</code>.<br />
+            It looks like this job hasn&apos;t been submitted from this browser.
+            Please ask the person that submitted this job for this information.
+          </p>
+        </section>
+      );
+    }
+    return <ErrorHandler />;
   }
 
+  const inputParameters =
+    'parameters' in inputParamsData
+      ? inputParamsData.parameters
+      : inputParamsData.data;
+
   return (
     <>
-      <ResubmitButton
-        inputParamsData={inputParamsData.data}
-        jobType={jobType}
-      />
+      {inputParameters && (
+        <ResubmitButton inputParamsData={inputParameters} jobType={jobType} />
```

Comment:
For consistency with blast/align what do you think about also listing this in the overview tab

Reply:
I think it might be good, yes, but we'll need to mess up with the results buttons that is used in a lot of other places, so I'll add it as a sticky and if we can't get to it soon as a jira